spec/Appendix A -- Persisted Documents.md

```diff
@@ -0,0 +1,256 @@
+lower-case hexidecimal encoding of the SHA256 hash (as specified in
+[RFC4634](https://datatracker.ietf.org/doc/html/rfc4634)) of the Source Text of
+the GraphQL Document (as specified in
+[the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language)).
+
+A service which accepts a _persisted document request_ SHOULD support the
+_SHA256 hex document identifier_ for compatibility.
+
+#### Example
+
+The following GraphQL query (with no trailing newline):
+
+```graphql example
+query ($id: ID!) {
+  user(id: $id) {
+    name
+  }
+}
+```
+
+Would have the following _SHA256 hex document identifier_:
+
+```example
+sha256:7dba4bd717b41f10434822356a93c32b1fb4907b983e854300ad839f84cdcd6e
+```
+
+Whereas the same query with all optional whitespace omitted:
+
+```raw graphql example
+query($id:ID!){user(id:$id){name}}
+```
+
+Would have this different _SHA256 hex document identifier_:
+
+```example
+sha256:71f7dc5758652baac68e4a10c50be732b741c892ade2883a99358f52b555286b
+```
+
+### Custom Document Identifier
+
+:: A _custom document identifier_ is a document identifier that contains no
+colon symbols (`:`). The meaning of a custom document identifier is
+implementation specific.
+
+Note: A 32 character hexidecimal _custom document identifier_ is likely to be an
+MD5 hash of the GraphQL document, as traditionally used by Relay.
+
+## Persisting a Document
+
+A client utilising persisted documents MUST generate a _document identifier_ for
+each GraphQL Document it wishes to issue to the server, and SHOULD ensure that
+the server can retrieve this GraphQL Document given the document identifier.
+
+Note: The method through which the client and server achieve this is
+implementation specific. Typically persisted documents are stored into some kind
+of trusted shared key-value store at client build time (either directly, or
+indirectly via an authenticated request to the server) such that the server may
+retrieve them given the identifier at request time. It is generally good
+practice for the client to issue both the GraphQL Document and the document
+identifier to the server; the server should then regenerate the document
+identifier from the GraphQL Document independently, and check that the
+identifiers match before storing the Document. An alternative approach has the
+client issue the GraphQL Document to the server, and the server returns an
+arbitrary _custom document identifier_ that the client should incorporate into
+its bundle.
+
+When using persisted documents as an operation allowlist, the client MUST
+persist the documents it uses ahead of time in a secure manner (preventing
+untrusted third parties from adding their own persisted document) such that the
+server will be able to retrieve the identified document within a _persisted
+document request_ and know that it is trusted.
+
+## Persisted Document Request
+
+A server MAY accept a _persisted document request_ via `GET` or `POST`.
```

Comment:
Do we define an opinion on mutations have to be POST to help folks avoid the pitfall of caching a mutation by accident

Reply:
You know; I figured this was already stated in the main spec so didn't need to be repeated... But you're right - it should be repeated. I'll add it.